harness/determined/exec/worker_process.py

```diff
@@ -58,4 +58,8 @@ def main() -> None:
 
 
 if __name__ == "__main__":
-    main()
+    try:
+        main()
+    except det.InvalidHP:
+        logging.info("InvalidHP detected, gracefully exiting trial")
```

Comment:
maybe reword this to not say the whole trial is exiting?

Reply:
Well the whole trial is exiting, but the experiment is continuing?

Reply:
Only the worker is exiting when this gets printed.  If you have 64 workers it makes sense to see 64 'worker exiting' messages, but not to see 64 "gracefully exiting trial messages", only to then see a decidedly non-graceful exit occur.